monkey/tests/unit_tests/infection_monkey/credential_collectors/test_mimikatz_collector.py

```diff
@@ -110,3 +117,24 @@ def test_pypykatz_result_parsing_no_secrets(monkeypatch):
     collected_credentials = collect_credentials()
     assert len(collected_credentials) == 1
     assert collected_credentials == expected_credentials
+
+
+@pytest.fixture
+def event_queue() -> IEventQueue:
+    return PyPubSubEventQueue(Publisher())
+
+
+def test_pypykatz_credentials_stolen_event_published(
+    monkeypatch, event_queue, event_queue_subscriber
+):
+    for event_tag in MIMIKATZ_EVENT_TAGS:
+        event_queue.subscribe_tag(event_tag, event_queue_subscriber)
+
+    mimikatz_credential_collector = MimikatzCredentialCollector(event_queue)
+    monkeypatch.setattr(
+        "infection_monkey.credential_collectors.mimikatz_collector.pypykatz_handler", Mock()
+    )
+    mimikatz_credential_collector.collect_credentials()
+
+    assert event_queue_subscriber.call_count == 3
+    assert event_queue_subscriber.call_tags == MIMIKATZ_EVENT_TAGS
```

Comment:
Using a real event queue and subscriber is overkill here. You can just pass in a mock event queue and us the mock object's built-in capabilities.